.github/workflows/changelogs/changelogs.py

```diff
@@ -0,0 +1,639 @@
+                # changelog entry are also mentioned in the PR
+                if pr_validation:
+                    for t in trackers:
+                        if kind not in pr_trackers or t not in pr_trackers[kind]:
+                            # Tracker not mentioned in the PR
+                            issues.append(Issue(IssueType.WRONG_TRACKER.format(t[0]), entry.file, entry.line, \
+                                                entry.end_line))
+
+            # Check Bugzilla trackers via the API
+            issues.extend(self.validate_bsc(entry))
+
+        # Check if all the trackers mentioned in the
+        # PR are also mentioned in the changelogs
+        if pr_validation:
+            for kind, trackers in pr_trackers.items():
+                for t in trackers:
+                    if kind not in all_trackers or t not in all_trackers[kind]:
+                        issues.append(Issue(IssueType.MISSING_TRACKER.format(t[0])))
+
+        return issues
+
+    def validate(self, file_list: list[str]) -> list[type[Issue]]:
+        """Validates changelogs in the list of files"""
+
+        # Index the list of files by package
+        self.pkg_idx = self.get_pkg_index(file_list)
+        issues = []
+        entries = []
+        for pkg, files in self.pkg_idx.items():
+            # General checks (package/changelog file mismatch)
+            if not files["files"]:
+                # Changelog added but no file is modified
+                issues.append(Issue(IssueType.WRONG_CHLOG, package=pkg))
+            if not files["changes"]:
+                # Files are modified but no changelog file added
+                issues.append(Issue(IssueType.MISSING_CHLOG, package=pkg))
+
+            # Validate each changelog file and gather all the issues
+            for file in files["changes"]:
+                i, e = self.validate_chlog_file(file)
+                issues.extend(i)
+                entries.extend(e)
+
+        # Validate all the mentioned trackers if the tracker file is provided
+        if self.regex.trackers:
+            issues.extend(self.validate_trackers(entries))
+
+        return issues
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="Validate changelog entries for Uyuni PRs",
+                                     epilog="Uyuni project: <https://github.com/uyuni-project/uyuni>")
+
+    parser.add_argument("-v", "--verbose",
+                        action="store_true",
+                        help="enable verbose output")
+
+    parser.add_argument("-l", "--line-length",
+                        type=int,
+                        default=DEFAULT_LINE_LENGTH,
+                        help="maximum line length allowed in changelog files (default: 67)")
+
+    parser.add_argument("-t", "--tracker-file",
+                        help="tracker definitions XML document retrieved from the OBS/IBS API. Bypass tracker validation if not provided.")
```

Comment:
Please, keep in mind that we have some trackers in IBS not available in OBS.

Reply:
I compared them, and it seems there are only minor differences in between. Which one should we use? OBS for Uyuni, and IBS for SUSE Manager? or something else?

Reply:
> I compared them, and it seems there are only minor differences in between. Which one should we use? OBS for Uyuni, and IBS for SUSE Manager? or something else?

the easy way it could also be to switch between Uyuni / OBS and SUMA / IBS

Reply:
...but it's true that for  using IBS we will need authentication.  :thinking: 

in any case, IBS has all the  trackers available in OBS, if the  parsed tracker  is valid in  OBS it'svalid in IBS too.  let's start easy

Reply:
OBS also requires authentication 🤷🏻‍♂️

Reply:
There's nothing risky in the data so my guess is it's just the default way the API is designed. But let's ask and make sure of course.

Reply:
@deneb-alpha thanks for opening the OBS issue. IIUC from this [comment](https://github.com/openSUSE/open-build-service/issues/15312#issuecomment-1840322618), it's ok to upload it to our public repo.